app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java

```diff
@@ -122,372 +116,187 @@ public void run() {
-                                );*/
-                                dbacess.toEventLog(4,"Detected MWI SMS");
-                                MiscUtils.startPopUpInfo(tContext, 7);
-                            } else {
-                                Log.d(TAG, " Detected Sms already logged");
-                            }
+        CapturedSmsData setmsg = new CapturedSmsData();
+        String smstext = findSmsData(bufflines,null);
+        String num = findSmsNumber(bufflines,null);
 
-                        } else if("WAPPUSH".equals(SILENT_ONLY_TAGS[index].split("#")[1].trim())) {
-                            /*
-                                Wap Push in logcat shows no data only senders number
-                                TODO: data is probably in db content://raw/1?
-                             */
-                            CapturedSmsData setmsg = new CapturedSmsData();
-                            setmsg.setSenderNumber("unknown");
-                            setmsg.setSenderMsg("no data");
-
-                            int startindex = x - 2;
-                            int endindex = x + 3;
-                            /*  wap push port DestPort 0x0B84
-                             *  its usually at this index of +3 in array                             *
-                              * */
-                            //This is index on Samsungs is different for other phone makes
-                            if (progress[x + 3].contains("DestPort 0x0B84")) {
-                                Log.d(TAG, "WAPPUSH DETECTED");
-                                /* loop thru array to find number */
-                                if (endindex+3 <= progress.length) {
-                                    while (startindex < endindex)
-                                    {
-                                        if (progress[startindex].contains(DETECTION_PHONENUM_SMS_DATA[2].toString())) {
-                                            try {
-                                                // Looking for OrigAddr this is where sender number is
-                                                String number = progress[startindex].substring(
-                                                        progress[startindex].indexOf("OrigAddr")).replace(
-                                                        DETECTION_PHONENUM_SMS_DATA[2].toString(), "").trim();
-
-                                                setmsg.setSenderNumber(number);
-                                                break;
-                                            } catch (Exception ee) {
-                                                Log.e(TAG, "Error parsing number", ee);
-                                            }
-
-                                        }
-                                        startindex++;
-
-                                    }
-                                }
+        if(smstext == null){smstext = "null";}
```

Comment:
I'm commenting on lack of whitespase, it's different. Too much whitespace is wrong too.

This is unreadable and can be skipped easily skipped while reading code, looking for bugs, finding vulns. There are reasons why every good and accepted by industry language has its official and community-accepted coding guidelines. You can find those guidelines basically in every place that is worth looking at:

oracle http://www.oracle.com/technetwork/java/codeconvtoc-136057.html
google https://google.github.io/styleguide/javaguide.html
cert https://www.securecoding.cert.org/confluence/display/java/3.+Reliability
openjdk https://wiki.openjdk.java.net/display/HotSpot/StyleGuide
Company http://your.high.profit.company.com

Since like, 2000 every IDE has own code formatter that in Java projects in by default set to **globally known and recognized** java coding guidelines. There are exceptions like:

if you are writing in pure-Java environment, it's safe to write:
```
if(smstext == null)
    smstext = "null";
```

but if in your team are some Python/Ruby programers or sysadmin who contribute to codebase, it's better to write:
```
if(smstext == null) { 
    smstext = "null";
}
```
As it will be recognized and understood by everyone.

Think why those code formatters in IDEs were invented, why tools like astyle matter.